x/staking/keeper/grpc_query.go

```
@@ -0,0 +1,460 @@
+}
+
+// Validator queries validator info for given validator addr
+func (k Querier) Validator(c context.Context, req *types.QueryValidatorRequest) (*types.QueryValidatorResponse, error) {
+	if req == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "empty request")
+	}
+
+	if req.ValidatorAddr.Empty() {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid request")
+	}
+	ctx := sdk.UnwrapSDKContext(c)
+	validator, found := k.GetValidator(ctx, req.ValidatorAddr)
+	if !found {
+		return nil, status.Errorf(codes.NotFound, "validator %s not found", req.ValidatorAddr)
+	}
+
+	return &types.QueryValidatorResponse{Validator: validator}, nil
+}
+
+// ValidatorDelegations queries delegate info for given validator
+func (k Querier) ValidatorDelegations(c context.Context, req *types.QueryValidatorDelegationsRequest) (*types.QueryValidatorDelegationsResponse, error) {
+	if req == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "empty request")
+	}
+
+	if req.ValidatorAddr.Empty() {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid request")
+	}
+	var delegations []types.Delegation
+	ctx := sdk.UnwrapSDKContext(c)
+
+	store := ctx.KVStore(k.storeKey)
+	valStore := prefix.NewStore(store, types.DelegationKey)
+	res, err := query.FilteredPaginate(valStore, req.Req, func(key []byte, value []byte, accumulate bool) (bool, error) {
+		delegation, err := types.UnmarshalDelegation(k.cdc, value)
+		if err != nil {
+			return false, err
+		}
+		if !delegation.GetValidatorAddr().Equals(req.ValidatorAddr) {
+			return false, nil
+		}
+
+		if accumulate {
+			delegations = append(delegations, delegation)
+		}
+		return true, nil
+	})
+	if err != nil {
+		return nil, status.Errorf(
+			codes.NotFound,
+			"unable to find delegations for validator %d", req.ValidatorAddr)
+	}
+
+	if len(delegations) == 0 {
+		return nil, status.Errorf(
+			codes.NotFound,
+			"unable to find delegations for validator %d", req.ValidatorAddr)
+	}
```

I don't think we should return an error on empty responses